tasks/install_dist_pkgs.yml

```
@@ -84,6 +84,13 @@
   args:
     warn: false
 
+- name: create log destination directory
+  file:
+    path: "{{ nrinfragent_config['log_file'] | dirname }}"
+    state: directory
+    recurse: true
+  when: nrinfragent_state != 'absent'
```

This will result in an error when there is no `nrinfragent_config['log_file']` defined.

>The task includes an option with an undefined variable. The error was: Unable to look up a name or access an attribute in template string ({{ nrinfragent_config['log_file'] | dirname }}). Make sure your variable name does not contain invalid characters like '-': expected str, bytes or os.PathLike object, not AnsibleUndefined

```suggestion
  when:
    - nrinfragent_state != 'absent'
    - nrinfragent_config['log_file'] is defined
```

---

Good catch, thanks for pointing it out!

---

The final state looks good to me, I've tested multiple cases. 👍